### PR TITLE
Deprecate old remote store

### LINKED ISSUE #32577: Deprecate old remote store if persistent sessions performs good enough for A/P
* Introduce a new feature "remote Infinispan supported" which is then deprecated
* On Keycloak startup check the CacheManager if remote store in cache XML is configured in any cache. If the feature is not enabled, don't start Keycloak
* Write release notes and migration guide -> point people to persistent sessions for user sessions. No replacement for authentication session or other sessions in remote store. Also note that remote store was never supported, except in Keycloak 23+24 when used with multi-site setups.

### PR DESCRIPTION
**Pull request by pruivo**
Closes #32577

### CODE REVIEW

**Comment by ahus1** [comment]
@pruivo - I added some migration guide. Can you please check? 

I found that when we added the CLI options for the remote cache, and people might have have used them independently from multi-site setups. We could have been more accurate on that in the docs :-(

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc [hunk 1]
```diff
@@ -111,14 +111,25 @@ As a consequence of the above changes, the following changes are required to you
...
+
+In {project_name} versions 24 to 25 to achieve a multi-site setups, a remote store in embedded Infinispan caches was configured. Manual configurations for the cache XML for multi-site were discouraged as CLI options to configure the caches are available.
+With the upgraded multi-site feature in 26, the same CLI options exist, but don't use embedded Infinispan caches anymore.
+
+In the community, remote stores for embedded Infinispan caches were used in some setups to keep user sessions when the {project_name} cluster was shut down or upgraded. This was never supported, documented or tested. As a fully supported alternative, the persistent user sessions feature should be used instead.
+
+Due to this, using remote caches in embedded Infinispan caches is now marked as deprecated with the plan to remove it.
+As it was never used outside multi-site setups that now achieve this by different means, it might be removed even in a future minor release.
+
+There is an experimental feature remote-cache which allows leveraging the new multi-site mechanisms to store session related data in an external {jdgserver_name} server also for non-multi-site setups.
```
Comment (pruivo) [inline docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc:131]: @ahus1 do we want to mention "also for non-multi-site setups"? Aren't we giving the green light to use in single-site setups?

Comment (ahus1) [inline docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc:131]:
```suggestion
There is an experimental feature remote-cache which allows leveraging the new multi-site mechanisms to store session related data in an external {jdgserver_name} server also for single site setups.
```

**Review by ahus1 — DISMISSED** [review]
Thank you for the change. We agreed in the call that we'll go with the text that people will try out the remote cache feature in single sites as the current implementation is agnostic of running in one or multiple sites.

**Review by ahus1 — APPROVED** [review]

**Pull request merged**